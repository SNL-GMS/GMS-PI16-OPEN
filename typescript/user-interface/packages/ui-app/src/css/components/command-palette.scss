.command-palette-overlay__contents {
  width: 33vw;
  max-width: 40rem;
  padding: 0.5em;
}
.command-palette {
  &__input {
    flex-grow: 1;
    border: 1px solid transparent;
    background: $gms-recessed;
    color: $gms-main;

    & input {
      font-size: 1rem;
    }
  }

  &__results-wrapper {
    max-height: max(50vh, 25em);
    overflow: auto;
    margin: 0 calc(-0.5em + 1px);
  }

  &__result-list {
    padding: 0;
    margin: 0;
    display: flex;
    flex-direction: column;
  }

  &__result {
    flex-grow: 1;
    background: $gms-prominent-background;
    padding: 0.5em;
    margin: 1px 0;
    list-style-type: none;

    &:first-of-type {
      margin-top: 0.5em;
    }

    &:hover {
      @include brighten;
    }

    &--selected {
      background-color: $gms-table-selection;
    }
  }
  & .is-highlighted {
    background-color: rgba($gms-main, 0.15);
  }
}
