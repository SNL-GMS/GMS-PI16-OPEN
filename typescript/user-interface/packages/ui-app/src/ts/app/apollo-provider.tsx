/* eslint-disable react/jsx-props-no-spreading */
import { IS_MODE_IAN } from '@gms/common-util';
import {
  ApolloClientConfiguration,
  Client,
  createApolloClientConfiguration,
  createApolloClientLogger,
  UILogger
} from '@gms/ui-apollo';
import { AppState } from '@gms/ui-state';
import { Actions } from '@gms/ui-state/lib/state/user-session/actions';
import React from 'react';
import { ApolloProvider } from 'react-apollo';
import * as Redux from 'redux';

/**
 * The disconnect grace period. This time must pass in order
 * to be considered truly disconnected. This is to help prevent
 * false or temporary disconnections/reconnects with the subscription.
 */
const DISCONNECT_GRACE_PERIOD_MS = 5000;

/**
 * The disconnected timeout.
 */
let disconnectedTimeout: NodeJS.Timeout;

/**
 * Wraps the provided component with Redux Provider and ApolloProvider.
 *
 * @param Component the component
 * @param store the redux store
 */
export const withApolloProvider = (
  Component: React.ComponentClass | React.FunctionComponent,
  store: Redux.Store<AppState>
): React.ComponentClass =>
  class WithApolloProvider extends React.PureComponent<any, any> {
    private static readonly apolloClientConfiguration: ApolloClientConfiguration = createApolloClientConfiguration();

    private static readonly apolloClientLogger: Client = createApolloClientLogger();

    public constructor(props) {
      super(props);
      UILogger.Instance().setClient(WithApolloProvider.apolloClientLogger);

      this.registerWsClientEvents();
    }

    /**
     * Wrap the component in an apollo and redux providers
     */
    // eslint-disable-next-line react/sort-comp
    public render(): JSX.Element {
      return (
        <ApolloProvider client={WithApolloProvider.apolloClientConfiguration.client}>
          <Component {...this.props} />
        </ApolloProvider>
      );
    }

    /**
     * Register websocket event handlers.
     */
    private readonly registerWsClientEvents = () => {
      // on disconnect; update the Redux connected status not connected
      WithApolloProvider.apolloClientConfiguration.wsClient.on('disconnected', () => {
        clearInterval(disconnectedTimeout);
        // set the grace period timeout; if this timeout fires then we are truly disconnected
        UILogger.Instance().debug(
          `Subscription disconnecting... waiting grace period to allow connection to reestablish...`
        );
        disconnectedTimeout = setInterval(() => {
          UILogger.Instance().debug(`Subscription disconnected...`);
          // TODO: Once there is a RIG client we are connected to, we should set this to false on disconnect.
          store.dispatch(Actions.setConnected(IS_MODE_IAN));
        }, DISCONNECT_GRACE_PERIOD_MS);
      });

      // on reconnected; update the Redux connected status to connected
      WithApolloProvider.apolloClientConfiguration.wsClient.on('reconnected', () => {
        // clear out the disconnect timeout; connection has been reestablished
        clearTimeout(disconnectedTimeout);
        disconnectedTimeout = undefined;
        UILogger.Instance().debug(`Subscription reconnected...`);
        store.dispatch(Actions.setConnected(true));
      });
    };
  };
