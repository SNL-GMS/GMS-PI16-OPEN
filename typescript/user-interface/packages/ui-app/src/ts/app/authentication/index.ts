import { IS_MODE_IAN } from '@gms/common-util';

import { ianAuthenticator } from './ian-authentication';
import { sohAuthenticator } from './soh-authentication';
import { Authenticator } from './types';

export const authenticator: Authenticator = IS_MODE_IAN ? ianAuthenticator : sohAuthenticator;
