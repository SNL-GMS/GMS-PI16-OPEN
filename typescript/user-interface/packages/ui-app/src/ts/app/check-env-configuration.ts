import {
  CESIUM_OFFLINE,
  GMS_UI_MODE,
  GRAPHQL_PROXY_URI,
  IS_MODE_IAN,
  IS_MODE_LEGACY,
  IS_MODE_SOH,
  IS_NODE_ENV_DEVELOPMENT,
  IS_NODE_ENV_PRODUCTION,
  NODE_ENV,
  SUBSCRIPTIONS_PROXY_URI,
  UI_URL,
  WAVEFORMS_PROXY_URI
} from '@gms/common-util';
import { UILogger } from '@gms/ui-apollo';

/**
 * Checks the ENV Configuration.
 * Logs the current environment configuration.
 */
export const checkEnvConfiguration = (): void => {
  UILogger.Instance().debug(
    `Environment (process.env): ` +
      `\n   process.env.NODE_ENV=${process.env.NODE_ENV}` +
      `\n   process.env.GMS_UI_MODE=${process.env.GMS_UI_MODE}` +
      `\n   process.env.GRAPHQL_PROXY_URI=${process.env.GRAPHQL_PROXY_URI}` +
      `\n   process.env.WAVEFORMS_PROXY_URI=${process.env.WAVEFORMS_PROXY_URI}` +
      `\n   process.env.SUBSCRIPTIONS_PROXY_URI=${process.env.SUBSCRIPTIONS_PROXY_URI}` +
      `\n   process.env.RIG_URI=${process.env.RIG_URI}` +
      `\n   process.env.CESIUM_OFFLINE=${process.env.CESIUM_OFFLINE}`
  );

  UILogger.Instance().debug(
    `App Environment: ` +
      `\n   NODE_ENV=${NODE_ENV}` +
      `\n   GMS_UI_MODE=${GMS_UI_MODE}` +
      `\n   IS_MODE_LEGACY=${IS_MODE_LEGACY}` +
      `\n   IS_MODE_IAN=${IS_MODE_IAN}` +
      `\n   IS_MODE_SOH=${IS_MODE_SOH}` +
      `\n   IS_NODE_ENV_DEVELOPMENT=${IS_NODE_ENV_DEVELOPMENT}` +
      `\n   IS_NODE_ENV_PRODUCTION=${IS_NODE_ENV_PRODUCTION}` +
      `\n   GRAPHQL_PROXY_URI=${GRAPHQL_PROXY_URI}` +
      `\n   WAVEFORMS_PROXY_URI=${WAVEFORMS_PROXY_URI}` +
      `\n   SUBSCRIPTIONS_PROXY_URI=${SUBSCRIPTIONS_PROXY_URI}` +
      `\n   UI_URL=${UI_URL}` +
      `\n   CESIUM_OFFLINE=${CESIUM_OFFLINE}`
  );
};
