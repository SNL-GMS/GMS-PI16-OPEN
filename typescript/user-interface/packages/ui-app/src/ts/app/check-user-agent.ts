import { UILogger } from '@gms/ui-apollo';

/**
 * Checks the user agent.
 */
export const checkUserAgent = (): void => {
  if (navigator) {
    // log user client info to the gateway
    UILogger.Instance().info(`client connected ${navigator.userAgent.toLowerCase()}`);
  }

  if (
    !window.navigator.userAgent.includes('Chrome') &&
    !window.navigator.userAgent.includes('Firefox')
  ) {
    // eslint-disable-next-line no-alert
    window.alert(`GMS Interactive Analysis currently supports
            Google Chrome > v59 and Firefox > v. You will likely experience degraded performance`);
  }
};
