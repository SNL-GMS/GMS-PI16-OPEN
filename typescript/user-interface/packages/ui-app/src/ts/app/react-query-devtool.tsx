/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import { deleteCookie, isCookieSet, setCookie } from '@gms/ui-util';
import * as React from 'react';
import { ReactQueryDevtoolsPanel } from 'react-query-devtools';

import { CommandPaletteContext } from '~components/common-ui/components/command-palette/command-palette-context';
import {
  Command,
  CommandScope,
  CommandType
} from '~components/common-ui/components/command-palette/types';

const DEV_TOOLS_COOKIE_ID = 'gms-react-query-devtool-status';

/**
 * Configures the React Query Dev Tool.
 * ONLY ENABLED FOR NODE_ENV === 'development' or MOCK === 'true' or SHOW_DEVTOOLS === 'true'
 */
export const ReactQueryDevtool: React.FunctionComponent<{
  shouldRender: boolean;
}> = props => {
  const initialState = isCookieSet(DEV_TOOLS_COOKIE_ID);
  const [isOpen, setIsOpen] = React.useState(initialState);
  const toggleDevTools = React.useCallback(() => {
    const current = isCookieSet(DEV_TOOLS_COOKIE_ID);
    if (!current) {
      // we will be opening it, so set the cookie
      setCookie(DEV_TOOLS_COOKIE_ID, 'isOpen');
    } else {
      deleteCookie(DEV_TOOLS_COOKIE_ID, 'isOpen');
    }
    setIsOpen(currentState => !currentState);
  }, []);
  // eslint-disable-next-line @typescript-eslint/unbound-method
  const { registerCommands } = React.useContext(CommandPaletteContext);
  const devToolsCommands: Command[] = React.useMemo(
    () => [
      {
        commandType: CommandType.SHOW_REACT_QUERY_DEV_TOOLS,
        // A list of strings that should be searched for this command
        searchTags: ['dev', 'query', 'mutation', 'tools', 'devtools'],
        // the function to call when the command is executed.
        action: toggleDevTools
      }
    ],
    [toggleDevTools]
  );

  const commandSignature = devToolsCommands.map(c => c.displayText).join();
  React.useEffect(() => {
    if (props.shouldRender) {
      registerCommands(devToolsCommands, CommandScope.DEV);
    }
  }, [commandSignature, devToolsCommands, props.shouldRender, registerCommands]);

  return isOpen && <ReactQueryDevtoolsPanel />;
};
