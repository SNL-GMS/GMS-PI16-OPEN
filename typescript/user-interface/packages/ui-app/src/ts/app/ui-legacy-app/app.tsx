/* eslint-disable @typescript-eslint/no-explicit-any */
import { Intent, NonIdealState, Spinner } from '@blueprintjs/core';
import { IconNames } from '@blueprintjs/icons';
import { IS_MODE_SOH } from '@gms/common-util';
import { AppState } from '@gms/ui-state';
import React from 'react';
import { Provider } from 'react-redux';
import { HashRouter, Route, Switch } from 'react-router-dom';
import * as Redux from 'redux';

import { AnalystUiComponents } from '~analyst-ui/';
import { InteractionWrapper } from '~analyst-ui/interactions/interaction-wrapper';
import { withApolloProvider } from '~app/apollo-provider';
import { authenticator } from '~app/authentication';
import { withReduxProvider } from '~app/redux-provider';
import { CommonUIComponents } from '~components/common-ui';
import { CommonCommandRegistrar } from '~components/common-ui/commands';
import { CommandPalette } from '~components/common-ui/components/command-palette';
import { wrapSystemMessageSubscription } from '~components/common-ui/components/system-message/system-message-subscription';
import { wrapSohStatusSubscriptions } from '~components/data-acquisition-ui/client-interface/apollo/soh-status-subscription-wrapper';
import { SohCommandRegistrar } from '~components/data-acquisition-ui/commands';
import { LoadingScreen } from '~components/loading-screen';
import { LoginScreen } from '~components/login-screen';
import { ProtectedRoute } from '~components/protected-route';
import { GoldenLayoutContext, Workspace } from '~components/workspace';
import { DataAcquisitionComponents } from '~data-acquisition-ui/';

import { createPopoutComponent } from '../create-popout-component';
import { glContextData } from './golden-layout-config';

/**
 * Wraps the component route (not for SOH).
 * Provides the required context providers to the component.
 *
 * @param Component the component route
 * @param props the props passed down from the route to the component
 * @param store the redux store
 * @param suppressPopinIcon true to force suppress the golden-layout popin icon
 */
const wrap = (
  Component: any,
  props: any,
  store: Redux.Store<AppState>,
  suppressPopinIcon = false
) => {
  const InteractionComp = InteractionWrapper(Component);
  const WrappedComponent: React.FunctionComponent = p => (
    <CommandPalette>
      {/* eslint-disable-next-line react/jsx-props-no-spreading */}
      <InteractionComp {...p} />
      <CommonCommandRegistrar />
    </CommandPalette>
  );

  return createPopoutComponent(
    withReduxProvider(
      withApolloProvider(wrapSystemMessageSubscription(WrappedComponent, props), store),
      store
    ),
    props,
    suppressPopinIcon
  );
};

/**
 * Wraps the component route for SOH.
 * Provides the required context providers to the component.
 *
 * @param Component the component route
 * @param props the props passed down from the route to the component
 * @param store the redux store
 * @param suppressPopinIcon true to force suppress the golden-layout popin icon
 */
const wrapSoh = (
  Component: any,
  props: any,
  store: Redux.Store<AppState>,
  suppressPopinIcon = false
) => {
  const InteractionComp = InteractionWrapper(Component);
  const WrappedComponent: React.FunctionComponent = p => (
    <CommandPalette>
      <CommonCommandRegistrar />
      <SohCommandRegistrar />
      {/* eslint-disable-next-line react/jsx-props-no-spreading */}
      <InteractionComp {...p} />
    </CommandPalette>
  );
  return createPopoutComponent(
    withReduxProvider(
      withApolloProvider(
        wrapSystemMessageSubscription(
          wrapSohStatusSubscriptions(WrappedComponent, props, store),
          props
        ),
        store
      ),
      store
    ),
    props,
    suppressPopinIcon
  );
};

export const App = (store: Redux.Store<AppState>): React.ReactElement => {
  const app: React.ReactElement = !IS_MODE_SOH ? (
    <NonIdealState
      icon={IconNames.ERROR}
      action={<Spinner intent={Intent.DANGER} />}
      title="Invalid settings"
      description="Not configured for ANALYST mode - Please check settings"
    />
  ) : (
    <Provider store={store}>
      <HashRouter>
        {
          // ! CAUTION: when changing the route paths
          // The route paths must match the `golden-layout` component name for popout windows
          // For example, the component name `signal-detections` must have the route path of `signal-detections`
        }
        {
          // For performance use `render` which accepts a functional component
          // that won't get unnecessarily remounted like with component.
        }
        <Switch>
          {
            // Authentication
          }
          <Route path="/login" render={() => <LoginScreen authenticator={authenticator} />} />
          <ProtectedRoute path="/loading" render={() => <LoadingScreen />} />
          {
            // Common UI
          }
          <ProtectedRoute
            path="/system-messages"
            render={props => wrap(CommonUIComponents.SystemMessage, props, store)}
          />
          {
            // Analyst UI
          }
          <ProtectedRoute
            path="/azimuth-slowness"
            render={props => wrap(AnalystUiComponents.AzimuthSlowness, props, store)}
          />
          <ProtectedRoute
            path="/events"
            render={props => wrap(AnalystUiComponents.Events, props, store)}
          />
          <ProtectedRoute
            path="/location"
            render={props => wrap(AnalystUiComponents.Location, props, store)}
          />
          <ProtectedRoute
            path="/magnitude"
            render={props => wrap(AnalystUiComponents.Magnitude, props, store)}
          />
          <ProtectedRoute
            path="/map"
            render={props => wrap(AnalystUiComponents.Map, props, store)}
          />
          <ProtectedRoute
            path="/signal-detections"
            render={props => wrap(AnalystUiComponents.SignalDetections, props, store)}
          />
          <ProtectedRoute
            path="/history"
            render={props => wrap(AnalystUiComponents.History, props, store)}
          />
          {
            // Data Acquisition
          }
          <ProtectedRoute
            path="/soh-overview"
            render={props => wrapSoh(DataAcquisitionComponents.SohOverview, props, store)}
          />
          <ProtectedRoute
            path="/station-statistics"
            render={props => wrapSoh(DataAcquisitionComponents.StationStatistics, props, store)}
          />
          <ProtectedRoute
            path="/soh-environment"
            render={props => wrapSoh(DataAcquisitionComponents.SohEnvironment, props, store)}
          />
          <ProtectedRoute
            path="/soh-missing"
            render={props => wrapSoh(DataAcquisitionComponents.SohMissing, props, store)}
          />
          <ProtectedRoute
            path="/soh-lag"
            render={props => wrapSoh(DataAcquisitionComponents.SohLag, props, store)}
          />
          {
            // Workspace
          }
          <GoldenLayoutContext.Provider value={glContextData(store)}>
            <ProtectedRoute
              path="/"
              render={props => wrapSoh(InteractionWrapper(Workspace), props, store, true)}
            />
          </GoldenLayoutContext.Provider>
        </Switch>
      </HashRouter>
    </Provider>
  );
  return app;
};
