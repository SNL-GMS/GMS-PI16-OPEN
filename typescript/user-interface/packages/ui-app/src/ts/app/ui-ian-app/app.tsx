/* eslint-disable @typescript-eslint/no-explicit-any */
import { Intent, NonIdealState, Spinner } from '@blueprintjs/core';
import { IconNames } from '@blueprintjs/icons';
import { IS_MODE_IAN, SHOW_DEVTOOLS } from '@gms/common-util';
import { AppState } from '@gms/ui-state';
import * as React from 'react';
import { Provider } from 'react-redux';
import { HashRouter, Route, Switch } from 'react-router-dom';
import * as Redux from 'redux';

import { AnalystUiComponents } from '~analyst-ui/';
import { IANMap, StationPropertiesComponent } from '~analyst-ui/components';
import { WaveformClientInitializer } from '~analyst-ui/components/waveform/waveform-client/waveform-client-initializer';
import { authenticator } from '~app/authentication';
import { ReactQueryDevtool } from '~app/react-query-devtool';
import { withReactQueryProvider } from '~app/react-query-provider';
import { withReduxProvider } from '~app/redux-provider';
import { ReduxStationsVisibilityInitializer } from '~app/redux-station-visibility-initializer';
import { InteractionWrapper } from '~components/analyst-ui/interactions/interaction-wrapper';
import { CommonCommandRegistrar } from '~components/common-ui/commands';
import { CommandPalette } from '~components/common-ui/components/command-palette';
import { LoadingScreen } from '~components/loading-screen';
import { LoginScreen } from '~components/login-screen';
import { ProtectedRoute } from '~components/protected-route';
import { GoldenLayoutContext, Workspace } from '~components/workspace';
import { WorkspaceProps } from '~components/workspace/types';

import { createPopoutComponent } from '../create-popout-component';
import { glContextData } from './golden-layout-config';

/**
 * Wraps the component route (not for SOH).
 * Provides the required context providers to the component.
 *
 * @param Component the component route
 * @param props the props passed down from the route to the component
 * @param store the redux store
 * @param suppressPopinIcon true to force suppress the golden-layout popin icon
 */
function wrap<T>(
  Component: any,
  props: any,
  store: Redux.Store<AppState>,
  suppressPopinIcon = false
) {
  const WrappedComponent: React.FunctionComponent<T> = p => {
    const InteractionComp = InteractionWrapper(Component);
    return (
      <>
        <CommandPalette>
          <ReactQueryDevtool shouldRender={SHOW_DEVTOOLS} />
          {/* eslint-disable-next-line react/jsx-props-no-spreading */}
          <InteractionComp {...p} />
          <CommonCommandRegistrar />
        </CommandPalette>
        <ReduxStationsVisibilityInitializer />
        <WaveformClientInitializer />
      </>
    );
  };

  return createPopoutComponent(
    withReduxProvider(withReactQueryProvider<T>(WrappedComponent), store),
    props,
    suppressPopinIcon
  );
}

export const App: React.FunctionComponent<{ store: Redux.Store<AppState> }> = ({
  store
}): React.ReactElement => {
  const app: React.ReactElement = !IS_MODE_IAN ? (
    <NonIdealState
      icon={IconNames.ERROR}
      action={<Spinner intent={Intent.DANGER} />}
      title="Invalid settings"
      description="Not configured for IAN mode - Please check settings"
    />
  ) : (
    <Provider store={store}>
      <HashRouter>
        {
          // ! CAUTION: when changing the route paths
          // The route paths must match the `golden-layout` component name for popout windows
          // For example, the component name `my-route` must have the route path of `my-route`
        }
        {
          // For performance use `render` which accepts a functional component
          // that won't get unnecessarily remounted like with component.
        }
        <Switch>
          {
            // Authentication
          }
          <Route path="/login" render={() => <LoginScreen authenticator={authenticator} />} />
          <ProtectedRoute path="/loading" render={() => <LoadingScreen />} />
          {
            // Individual Routes
          }
          <ProtectedRoute
            path="/workflow"
            render={props => wrap(AnalystUiComponents.Workflow, props, store)}
          />
          <ProtectedRoute
            path="/waveform-display"
            render={props => wrap(AnalystUiComponents.WaveformDisplay, props, store)}
          />
          <ProtectedRoute path="/map-display" render={props => wrap<never>(IANMap, props, store)} />
          <ProtectedRoute
            path="/station-properties"
            render={props => wrap<never>(StationPropertiesComponent, props, store)}
          />
          {
            // Workspace
          }
          <GoldenLayoutContext.Provider value={glContextData(store)}>
            <ProtectedRoute
              path="/"
              render={props => wrap<WorkspaceProps>(Workspace, props, store, true)}
            />
          </GoldenLayoutContext.Provider>
        </Switch>
      </HashRouter>
    </Provider>
  );
  return app;
};
