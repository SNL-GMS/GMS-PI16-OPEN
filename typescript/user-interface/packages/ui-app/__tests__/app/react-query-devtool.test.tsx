import React from 'react';

import { ReactQueryDevtool } from '../../src/ts/app/react-query-devtool';
import {
  Command,
  CommandPaletteContext,
  CommandType
} from '../../src/ts/components/common-ui/components/command-palette';
import { CommandPaletteContextData } from '../../src/ts/components/common-ui/components/command-palette/command-palette-context';

// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
const Enzyme = require('enzyme');

describe('react query devtool', () => {
  let toggleOpenCommand: Command;
  const context: CommandPaletteContextData = {
    commandPaletteIsVisible: false,
    registerCommands: jest.fn().mockImplementation((commandsToRegister: Command[]) => {
      toggleOpenCommand = commandsToRegister.find(
        (c: Command) => c.commandType === CommandType.SHOW_REACT_QUERY_DEV_TOOLS
      );
    }),
    setCommandPaletteVisibility: jest.fn()
  };
  const Wrapper = Enzyme.mount(
    <CommandPaletteContext.Provider value={context}>
      <ReactQueryDevtool shouldRender />
    </CommandPaletteContext.Provider>
  );

  it('matches a snapshot', () => {
    expect(Wrapper).toMatchSnapshot();
  });

  it('sets a cookie to remember when opened', () => {
    toggleOpenCommand.action();
    expect(document.cookie).toBeTruthy();
  });

  it('removes a cookie to remember when closed', () => {
    expect(document.cookie).toBeTruthy();
    toggleOpenCommand.action();
    expect(document.cookie).toBeFalsy();
  });
});
