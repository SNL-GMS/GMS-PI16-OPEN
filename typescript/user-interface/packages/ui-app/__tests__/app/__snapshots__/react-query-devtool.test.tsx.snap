// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react query devtool matches a snapshot 1`] = `
<ReactQueryDevtool
  shouldRender={true}
/>
`;
